### DISCUSSION TITLE
Adding a new sensor

### DISCUSSION THREAD

**Discussion by m3sbelgium** [rodralez/NaveGo#64] [Ideas]
Hello Rodrigo,

I would like to add a wheel-based odometer sensor for the data fusion. 

What would be your advice regarding how to proceed easily, efficiently and in the most appropriate way ?

Best regards,
Jérémy

**Comment by rodralez** [comment]
Hi,

I don't have a previous experience in implementing a wheel-based odometer sensor in NaveGo. But, I think you can follow the same path I followed when I added a magnetometer to NaveGo (see file ins-gnss/ins_gnss_mag.m).

Firstly, you have to modify 1) the matrix H adding an extra row, 2) matrix R adding the noise profile of the odometer, and 3) vector z (see line 370 in ins_gnss_mag.m).

When you finish your project, we may add your work to NaveGo, if you agree.

Regards.
